modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/MuleExtensionAnnotationParser.java

```diff
@@ -154,9 +156,9 @@ public static Extension getExtension(Class<?> extensionType)
 
     private static void parseGroupParameters(MetadataType parameterType, List<ParsedParameter> parsedParameters, ClassTypeLoader typeLoader)
     {
-        for (Field field : IntrospectionUtils.getParameterFields(JavaTypeUtils.getType(parameterType)))
+        for (Field field : JavaTypeUtils.getType(parameterType).getDeclaredFields())
```

Comment:
Instead of grabbing all the fields, we can do a double filter, by @Parameter and @MetadataKeyPart